### ISSUE TITLE
Replicating CIFAR100-LT results

### ISSUE THREAD

**Issue by rahulvigneswaran** [naver-ai/cmo#4]
Hey, I ran the repo without any changes but was not able to replicate the results exactly as provided in the paper. 

| Method         | From Paper (Unknown seed) | Replicated (seed 1) | Replicated (seed 2) | Replicated (seed 3) |
|----------------|--------------------------:|--------------------:|--------------------:|--------------------:|
| CE             |                      38.6 |               38.81 |               38.46 |               38.22 |
| CE + CMO       |                      43.9 |               42.21 |               41.72 |                42.4 |
| BS + CMO       |                      46.6 |               46.53 |               46.03 |               45.64 |
| LDAM-DRW + CMO |                      47.2 |               46.06 |               46.45 |               46.49 |

A tolerance of +/- 0.2 is acceptable but in some cases, there seems to be a gap of around 2.18 which is not ideal.
| Method         | Difference from the results given in the paper (seed 1) | Difference from the results given in the paper (seed 2) | Difference from the results given in the paper (seed 3) |
|----------------|--------------------------------------------------------:|--------------------------------------------------------:|--------------------------------------------------------:|
| CE             |                                                    +0.21 |                                                   -0.14 |                                                   -0.38 |
| CE + CMO       |                                                   -1.69 |                                                   -2.18 |                                                    -1.5 |
| BS + CMO       |                                                   -0.07 |                                                   -0.57 |                                                   -0.96 |
| LDAM-DRW + CMO |                                                   -1.14 |                                                   -0.75 |                                                   -0.71 |

- [ ] Can you let me know where am going wrong here or if am missing something?
- [ ] Can you let me know the PyTorch version you are using?

**Comment by pseulki** [comment]
Hi,
Sorry for the late reply, and sorry to hear that you could not get the same results.

We did not set the seed to a specific number so that the method does not overfit to the specific seed setting.
Also, even with the same seed, I found out that the results remained stochastic.
For example, when I ran CE+CMO with seed=1, I got 43.13%, 43.46%, 42.6%.
This may be related to this issue: https://github.com/pytorch/pytorch/issues/7068

For tips, there are some hyperparameters in CMO, such as mixup_prob, start_data_aug, end_data_aug.
For CIFAR, due to limited number of dataset, increasing mixup_prob to 0.8 or 1. can produce more new images.

Lastly, my pytorch version is as follows:
pytorch: 1.6, torchvision: 0.7, numpy: 1.18

Thank you for your interest :)
